src/main/resources/public/ts/controllers/controller.ts

```diff
@@ -0,0 +1,788 @@
+						return ticket.id === id;
+					});
+				if (hasTicket.length === 0){
+					model.getTicket(ticketId, function(result) {
+						$scope.ticket = new Ticket(result[0]);
+						$scope.openTicket(id);
+					});
+				}else{
+					$scope.ticket = hasTicket[0];
+					$scope.openTicket(id);
+				}
+			}else{
+				notify.info('support.ticket.search.warning');
+			}
+		};
+
+		$scope.openTicket = function(id) {
+			if(!$scope.ticket || ($scope.ticket && $scope.ticket.id !== id))
+				$scope.ticket = _.find(model.tickets.all, function(ticket){
+					return ticket.id === id;
+				});
+			if(!$scope.ticket) {
+				if($scope.searchInput){
+					$scope.notFoundSearchInput = true;
+					$scope.searchInput = false;
+				}else{
+					$scope.notFound = true;
+				}
+				$scope.$apply();
+				return;
+			}
+			template.open('main', 'view-ticket');
+			$scope.ticket.getAttachments();
+			$scope.ticket.getComments(function() {
+				$scope.initHisto(id.toString());
+			});
+			model.getProfile($scope.ticket.owner, function(result) {
+				$scope.ticket.profile = result.profile;
+			});
+			$scope.ticket.getBugTrackerIssue();
+		};
+
+		$scope.viewTicket = function(ticketId) {
+			window.location.hash = '/ticket/' + ticketId;
+		};
+
+		// called when opening ticket and updating
+		$scope.initHisto = function(ticketId) {
+			model.getEvents(ticketId, function(result) {
+				template.open('histo-ticket', 'histo-ticket');
+				$scope.events = result;
+
+				$scope.events.forEach(function (event){
+					const comment: any = {};
+					comment.owner = event.user_id;
+					comment.owner_name = event.username;
+					comment.created = event.event_date;
+					comment.content = event.event;
+					comment.status = $scope.getStatusLabel(event.status);
+					comment.isHistory = true;
+					comment.type = event.event_type;
+					$scope.ticket.comments.push(comment);
+				});
+
+				// adding the comments from bug tracker.
+				if( $scope.isBugTrackerCommDirect ){
+					if( $scope.ticket.issue && $scope.ticket.issue.journals && $scope.ticket.issue.journals.length > 0 ) {
+						//get the bug tracker author name
+						$scope.bugTrackerAuthor = $scope.ticket.issue.author.name;
+						$scope.ticket.issue.journals.forEach( function( btComment) {
+							if( btComment.notes !== "" ) {
+								//alert(btComment.notes);
+								// alert(btComment.created_on);
+								if(btComment.user.name !== $scope.bugTrackerAuthor){
+									let comment: any = {};
+									comment.created = btComment.created_on;
+									comment.content = btComment.notes;
+									comment.type = 5;
+									comment.isHistory = true;
+									comment.owner_name = btComment.user.name;
+									$scope.ticket.comments.push(comment);
+								}
+							}
+						});
+					}
+				}
+
+				$scope.ticketHisto = ticketId;
+				$scope.$apply();
+				//$scope.ticketLib = ticket.subject;
+			}, function (e) {
+				$scope.processingData = false;
+				// validationError(e);
```

Comment:
Commenté car en javascript et typescript on a une erreur "ValidationError is not defined" (cf voir description PR)